packages/utils/src/getRandomInt.ts

```diff
@@ -0,0 +1,58 @@
+import { getRandomValues as cryptoGetRandomValues } from 'crypto';
+
+/**
+ * Before changing anything here, see the Modulo Bias problem!
+ * @see https://research.kudelskisecurity.com/2020/07/28/the-definitive-guide-to-modulo-bias-and-how-to-avoid-it/
+ *
+ * @param min Inclusive
+ * @param max Exclusive (to match the crypto.randomInt() function API
+ */
+export const getRandomInt = (min: number, max: number) => {
+    if (!Number.isSafeInteger(min)) {
+        throw new RangeError(
+            `The "min" argument must be a safe integer. Received type ${typeof min} (${min})`,
+        );
+    }
+
+    if (!Number.isSafeInteger(max)) {
+        throw new RangeError(
+            `The "max" argument must be a safe integer. Received type ${typeof max} (${max})`,
+        );
+    }
+
+    if (min >= max) {
+        throw new RangeError(
+            `The value of "max" is out of range. It must be greater than the value of "min" (${min}). Received ${max}`,
+        );
+    }
+
+    const MAX_32_BIT_INT = 0xffffffff;
+    const range = max - min;
+
+    if (range > MAX_32_BIT_INT) {
+        throw new RangeError(
+            `This function only provide 32 bits of entropy, therefore range cannot be more then 2^32.`,
+        );
+    }
```

Comment:
If you call `getRandomInt(0, 2^32)`, you'll get the exception: `This function only provide 32 bits of entropy, therefore range cannot be more then 2^32.` First, I think this is too restrictive, since we can easily handle this case as well. Second, the message in the exception is not true since the range is `2^32` which is not more than `2^32`.

If you change `MAX_32_BIT_INT` to `2^32` (and perhaps rename the variable), it should resolve both the problems. At the same time, it should not cause any issue elsewhere: It will hold that `0 <= maxRange <= 2^32` and the loop will generate evenly distributed  `randomValue` such that `0 <= randomValue < maxRange`.